README.md

```diff
@@ -382,17 +382,218 @@ registry.define('my-widget', () => {
 
 #### Theming
 
-// To be completed
+Widgets are themed using `css-modules` and the `themeable` mixin. Each widget should import it's own `baseTheme` and then apply the generate classes to the vdom it creates. Theme classes are acquired by calling `instance.theme.<themeClass>`.
+
+##### Themeable Mixin
+The Themeable mixin provides a widget with the `css-module` theme classes it needs to style it's components. It requires a `baseTheme` to be set and a generic to determine the typings for the returned `theme` classes. The `baseTheme` must contain a complete set of all of the classes you wish to apply to a widget as all theme and override classes are keyed off of it.
+
+``` typescript
+interface ThemeableMixin<P> extends Evented {
+	theme: AppliedClasses<P>;
+}
+
+interface ThemeableProperties {
+	theme?: {};
+	overrideClasses?: {};
+}
+
+interface Themeable<P> extends ThemeableMixin<P> {
+	baseTheme: P;
+	properties: ThemeableProperties;
+}
+```
+
+Usage:
+
+```  css
+// tabpanel.css
+.tabPanelTabs {
+	background: red;
+}
+
+.tabPanelTab {
+	background: blue;
+}
+```
+
+``` typescript
+// tabpanel.ts
+import * as baseTheme from './styles/tabpanel';
+import themeableMixin, { Themeable } from '../mixins/themeable';
+
+export type TabPanel = Widget<WidgetProperties> & Themeable<typeof baseTheme>;
+export interface TabPanelFactory extends ComposeFactory<TabPanel, WidgetProperties> {}
+
+const createTabPanel: TabPanelFactory = createWidgetBase.mixin(themeableMixin).mixin({
+	mixin: {
+		baseTheme,
+		getChildrenNodes: function (this: TabPanel): DNode[] {
+			return [
+				v(`ul`, { classes: this.theme.tabPanelTabs }, [
+					v('li', { classes: this.theme.tabPanelTab }, [ 'tab1' ])
+					// ...
+				]);
+			];
+		}
+	}
+});
+```
+
+##### Setting a theme
+Themeable properties includes an optional `theme` property which can be set to pass a theme to a widget. Theme classes will override `baseTheme` classes. When a `theme` property is set or changed, the widgets `theme` classes will be regenerated and the widget invalidated such that it is redrawn. Themes are used to apply consistent styling across the widget code base.
+
+Usage Extending on the previous `tabPanel` example.
+
+``` css
+// customTheme.css
+.tabPanelTabs {
+	background: green;
+}
+
+.notATabClass {
+	// this class will not be available in `instance.theme` as
+	// it is not part of the baseTheme
+	background: yellow;
+}
+```
+
+``` typescript
+import * as customTheme from './themes/customTheme';
+
+w(createTabPanel, { properties: { theme: customTheme } });
+// Resulting widget will have green tabs instead of baseTheme red.
+```
+
+##### Changing a theme
+A theme can be changed on a widget by using it's `setProperties` function.
+Example building on the above custom theme.
+
+``` typescript
+import * as customTheme from './themes/customTheme';
+
+const myTabPanel = w(createTabPanel, {});
+// myTabPanel will have red tabs.
+
+myTabPanel.setProperties({ theme: customTheme });
```

Comment:
I don't think we want to just take the `w` usage out, because we need to be clear about how and where the theme is actually set (root widget, i.e. projector)...